File: src/com/shape/ast/attributes/DimensionAttribute.java
package com.shape.ast.attributes;

import com.shape.dictionary.TokenType;
import com.shape.error.SyntaxError;
import com.shape.lexer.Token;

public class DimensionAttribute extends ResourceAttribute {
    private Integer valueI;
    private Double valueD;
    private Token type;

    public DimensionAttribute(String key, String resourceType, String resourceName) throws SyntaxError {
        super(key, resourceType, resourceName);
        if (!this.resourceType.equals("dimen")) {
            // TODO: raise syntax error if not @dimen
//          throw new SyntaxError("Only dimension resource");
        }
    }

    public DimensionAttribute(String key, Token value, Token typeToken) {
        super(key, ""); // will fill value later
        this.type = typeToken;

        if (value.getType() == TokenType.NUMBER) {
            this.valueI = Integer.parseInt(value.getLexeme());
            this.value = valueI + typeToken.getLexeme();

        } else if (value.getType() == TokenType.OLD_DOUBLE) {
            this.valueD = Double.parseDouble(value.getLexeme());
            this.value = valueD + typeToken.getLexeme();
        }
    }
}

File: compiler/src/main/java/com/shape/ast/attributes/DimensionAttribute.kt
package com.shape.ast.attributes

import com.shape.dictionary.TokenType
import com.shape.error.SyntaxError
import com.shape.lexer.Token

/**
 * Copyright (C) 2015 Therne. All rights are reserved.
 * @author Vista
 */
class DimensionAttribute : ResourceAttribute {
    var valueI: Int = 0
    var valueD: Double = 0.0
    val type: Token?

    constructor(key: String, value: Token, typeToken: Token) : super(key, "") {
        type = typeToken

        if (value.type == TokenType.NUMBER) {
            this.valueI = Integer.parseInt(value.lexeme)
            this.value = valueI.toString() + typeToken.lexeme

        } else if (value.type == TokenType.OLD_DOUBLE) {
            this.valueD = java.lang.Double.parseDouble(value.lexeme)
            this.value = valueD.toString() + typeToken.lexeme
        }
    }
}